awxkit/awxkit/api/pages/credentials.py

```diff
@@ -302,6 +303,10 @@ def create(
         return self.update_identity(
             Credentials(
                 self.connection)).post(payload)
+    
+    def test(self, data):
+        """Test the credential endpoint."""
+        return self.connection.post(urljoin(str(self.url), 'test/' ), data)
```

Comment:
This looks good. You could probably add another one of these to `CredentialType`. It could be used to check the endpoint used in this scenario:

> There is one exception - if you haven't saved a credential (e.g: you're on the 'NEW CREDENTIAL' form of a hashivault credential and haven't clicked save yet), then only admins may use the test button.

context: On the `NEW CREDENTIAL` page, we use `api/v2/credential_types/n/test`.